packages/core/src/render3/node_manipulation.ts

```diff
@@ -353,32 +384,39 @@ function cleanUpView(tView: TView, lView: LView): void {
   }
 }
 
-function hasLeaveAnimations(lView: LView | undefined): boolean {
-  return lView !== undefined && lView[ANIMATIONS] !== null && lView[ANIMATIONS].leave !== undefined;
-}
+function runLeaveAnimationsWithCallback(
+  lView: LView | undefined,
+  rNode: RNode,
+  injector: Injector,
+  callback: Function,
+) {
+  const animations = lView?.[ANIMATIONS];
+  if (animations == null) return callback(false);
 
-function runLeaveAnimationsWithCallback(lView: LView | undefined, callback: Function) {
-  if (lView && lView[ANIMATIONS] && lView[ANIMATIONS].leave) {
-    if (lView[ANIMATIONS].skipLeaveAnimations) {
-      lView[ANIMATIONS].skipLeaveAnimations = false;
+  addToAnimationQueue(injector, () => {
+    if (animations.skipLeaveAnimations) {
+      animations!.skipLeaveAnimations = false;
```

Comment:
nit: this should not be necessary if we've made it into this branch:
```suggestion
      animations.skipLeaveAnimations = false;
```